### PR TITLE
Add index exists function in index_scheduler which stops opening indexes to only check if they exist.

### LINKED ISSUE #4784: Stop opening indexes to only check if they exists
We must not open indexes to check if an index exists but rather use a new method to do that **without** loading the index in memory.

https://github.com/meilisearch/meilisearch/blob/05cc2d1fac7554b8210086ccccd524010511dc00/meilisearch/src/routes/indexes/documents.rs#L306-L308

### PR DESCRIPTION
**Pull request by Karribalu**
# Pull Request

## Related issue
Fixes #4784

## What does this PR do?
- Added index_exists function in the index_scheduler.
- Resolved opening indexes to only check if they exist.
- Made changes to existing tests to test this function.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: meilisearch/src/routes/indexes/documents.rs [hunk 1]
```diff
@@ -304,7 +304,11 @@ pub async fn replace_documents(
     debug!(parameters = ?params, "Replace documents");
     let params = params.into_inner();
 
-    analytics.add_documents(&params, index_scheduler.index(&index_uid).is_err(), &req);
+    analytics.add_documents(
+        &params,
+        !matches!(index_scheduler.index_exists(&index_uid), Ok(true)),
```
Comment (Kerollmops) [inline meilisearch/src/routes/indexes/documents.rs:309]:
```suggestion
        index_scheduler.index_exists(&index_uid) == Ok(false),
```

File: meilisearch/src/routes/indexes/documents.rs [hunk 2]
```diff
@@ -341,7 +345,11 @@ pub async fn update_documents(
     let params = params.into_inner();
     debug!(parameters = ?params, "Update documents");
 
-    analytics.update_documents(&params, index_scheduler.index(&index_uid).is_err(), &req);
+    analytics.update_documents(
+        &params,
+        !matches!(index_scheduler.index_exists(&index_uid), Ok(true)),
```
Comment (Kerollmops) [inline meilisearch/src/routes/indexes/documents.rs:350]:
```suggestion
        index_scheduler.index_exists(&index_uid) == Ok(false),
```

File: index-scheduler/src/lib.rs [hunk 3]
```diff
@@ -662,7 +662,11 @@ impl IndexScheduler {
         let rtxn = self.env.read_txn()?;
         self.index_mapper.index(&rtxn, name)
     }
-
+    /// Return the boolean referring if index exists.
+    pub fn index_exists(&self, name: &str) -> Result<bool> {
+        let rtxn = self.env.read_txn()?;
+        self.index_mapper.index_exists(&rtxn, name)
+    }
```
Comment (Kerollmops) [inline index-scheduler/src/lib.rs:669]: Ahah! I didn't remember we had this method 🤩

```suggestion
    pub fn index_exists(&self, name: &str) -> Result<bool> {
        let rtxn = self.env.read_txn()?;
        self.index_mapper.index_exists(&rtxn, name)
    }

```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for your PR @Karribalu 🃏
Can you change some small stuff, please?

**Comment by Karribalu** [comment]
@Kerollmops 
Done the changes, Please review and let me know of any changes

**Comment by Karribalu** [comment]
@Kerollmops 
I just realised, It should be 
index_scheduler.index_exists(&index_uid) != Ok(false)
right

File: meilisearch/src/routes/indexes/documents.rs [hunk 4]
```diff
@@ -304,7 +304,7 @@ pub async fn replace_documents(
     debug!(parameters = ?params, "Replace documents");
     let params = params.into_inner();
 
-    analytics.add_documents(&params, index_scheduler.index(&index_uid).is_err(), &req);
+    analytics.add_documents(&params, index_scheduler.index_exists(&index_uid) == Ok(true), &req);
```
Comment (Kerollmops) [inline meilisearch/src/routes/indexes/documents.rs:307]:
```suggestion
    analytics.add_documents(&params, index_scheduler.index_exists(&index_uid).map_or(false, |x| x), &req);
```

File: meilisearch/src/routes/indexes/documents.rs [hunk 5]
```diff
@@ -341,7 +341,7 @@ pub async fn update_documents(
     let params = params.into_inner();
     debug!(parameters = ?params, "Update documents");
 
-    analytics.update_documents(&params, index_scheduler.index(&index_uid).is_err(), &req);
+    analytics.update_documents(&params, index_scheduler.index_exists(&index_uid) == Ok(true), &req);
```
Comment (Kerollmops) [inline meilisearch/src/routes/indexes/documents.rs:344]:
```suggestion
    analytics.update_documents(&params, index_scheduler.index_exists(&index_uid).map_or(false, |x| x), &req);
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thanks @Karribalu! You could probably use the `std::convert::identity` instead of `|x| x` 🤔

File: meilisearch/src/routes/indexes/documents.rs (hunk 5 above)
Comment (Karribalu) [reply meilisearch/src/routes/indexes/documents.rs:344]: Yeah, That would work but I think we have to pass the negation of the response meaning 
if the index exist we have to pass false to the function, Correct me if I am wrong.

Comment (Kerollmops) [reply meilisearch/src/routes/indexes/documents.rs:344]: Ho! Yeah, you are right. It's exactly that, my bad 😅

Comment (Karribalu) [reply meilisearch/src/routes/indexes/documents.rs]: @Kerollmops 
Addressed the proposals and updated the PR.
Let me know of any issues

**Review by Kerollmops — APPROVED** [review]
Wonderful1 Thank you very much!
bors merge

**Pull request merged**